Config Split

samcorky opened the pull request:
Split config

samcorky committed:
Refactored configuration imports and splits

Mainly, the configuration imports in various modules have been refactored for better organization. Configurations have now been split into 'smib.slack.config' and 'smib.webserver.config' files, ensuring that the relevant configurations are loaded for each specific module. Docker build arguments have also been added to exclude unnecessary directories during the build process, making the build more efficient.

There is a core smib.common.config which holds configuration necessary to both parts of the app.

Docker compose file/build file have had an argument added to remove the other containers source, this is to stop an dependancy injection issue. Also makes the respective images a tad smaller

samcorky commented:
blocked by #68 

I should've based this branch off of that one

sjefferson99 commented:
@sam57719 Merge branch 68 into this one and keep it open. We can merge 68 into develop and then 69 on top later, or just merge 69 in and discard 68 as we need.

samcorky committed:
Remove 'exclude_dir' argument from Docker configurations

The Docker configuration files, docker-compose.yml and smib-fast.Dockerfile, have been modified to remove the 'exclude_dir' argument. This argument was previously used to exclude directories from the Docker build process, however, this exclusion was found unnecessary. Now, the Docker build workflow is simpler and does not exclude any directories.

This is now done in the load_injection_container calls to properly sort the injection issues

samcorky commented:
@sjefferson99  this now should be able to go in separatley to the docker branches #68 - found better way of sorting the injectable issue

sjefferson99 commented on docker-compose.yml:
@sam57719 Does this mean that env var export is no longer supported and tokens must now be provided with .env?

samcorky replied:
@sjefferson99 

ENV var will still be supported as it currently is - for any values that are set in the docker compose like this
```
    environment:
      ...

      # Passed in from HOST
      - SLACK_APP_TOKEN
      - SLACK_BOT_TOKEN
```

samcorky replied:
.env will be cleaner - will update readme :)

sjefferson99 replied:
Confirmed this is just removing from the webserver compose element, that no longer needs it.

sjefferson99 reviewed (approved):
All fine.
